dotnet/src/Plugins/Plugins.Web/Bing/BingConnector.cs

```diff
@@ -39,31 +43,30 @@ public sealed class BingConnector : IWebSearchEngineConnector
     /// </summary>
     /// <param name="apiKey">The API key to authenticate the connector.</param>
     /// <param name="httpClient">The HTTP client to use for making requests.</param>
+    /// <param name="uri"></param>
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
-    public BingConnector(string apiKey, HttpClient httpClient, ILoggerFactory? loggerFactory = null)
+    [SuppressMessage("Performance", "CA1054:Change the type of parameter 'uri'...",
+        Justification = "A constant Uri cannot be defined, as required by this class")]
+    public BingConnector(string apiKey, HttpClient httpClient, string uri = DefaultUri, ILoggerFactory? loggerFactory = null)
     {
         Verify.NotNull(httpClient);
 
         this._apiKey = apiKey;
         this._logger = loggerFactory is not null ? loggerFactory.CreateLogger(typeof(BingConnector)) : NullLogger.Instance;
         this._httpClient = httpClient;
         this._httpClient.DefaultRequestHeaders.Add("User-Agent", Telemetry.HttpUserAgent);
+        this._uri = string.IsNullOrEmpty(uri) ? BingConnector.DefaultUri : uri;
```

Comment:
You could use `Uri? uri` in the constructor and this line be `this._uri = uri ?? new Uri(DefaultUri)`

Reply:
@lemillermicrosoft a Uri? cannot be instantiated as a compile time constant. 

As a result this use would (does) throw errors, hence the approach I took as seen in commits for this PR.

![image](https://github.com/microsoft/semantic-kernel/assets/95053834/51b0954e-af78-48e2-ad72-95746030e4a6)

Reply:
The default in the constructor would be null instead.

Reply:
```
public BingConnector(string apiKey, Uri? uri = null, ILoggerFactory? loggerFactory = null)
```

Reply:
Ah Ok, from the PR previously, it had been mentioned that the constructor should reveal the DefaultUri being used (when hovered over).

Happy to make this change though (it's in https://github.com/microsoft/semantic-kernel/pull/3357/commits/485ca374c1af20d824b9f45f142395ca1e4be445)